frontend/src/models/archive.ts

```diff
@@ -0,0 +1,41 @@
+export interface Archive {
+    archiveID: number;
+    archiveURL: string;
+}
+
+export interface ArchiveJson {
+    archiveID: number;
+    archiveURL: string;
+}
+
+export function archiveFromJson(json: ArchiveJson): Archive {
+    const { archiveID, archiveURL } = json;
+    return {
+        archiveID,
+        archiveURL: lengthenArchiveURL(archiveURL),
+    };
+}
+
+export function archiveToJson(archive: Archive): ArchiveJson {
+    const { archiveID, archiveURL } = archive;
+    return {
+        archiveID,
+        archiveURL: shortenArchiveURL(archiveURL),
+    };
+}
+
+/**
+ * Turn a full youtube-nocookie link into the 11-character ID
+ * @param archiveURL
+ */
+function shortenArchiveURL(archiveURL: string): string {
+    return archiveURL.substring(39, 50);
+}
+
+/**
+ * Turn a shortened 11-character ID into a full youtube-nocookie link
+ * @param archiveURL 
+ */
+function lengthenArchiveURL(archiveURL: string): string {
+    return "https://www.youtube-nocookie.com/embed/" + archiveURL + "?rel=0";
```

Comment:
We should probably make a contributing.md that requires everybody to use youtube-nocookie over normal tracking youtube links.